src/main/kotlin/com/openlattice/data/storage/PostgresDataQueries.kt

```diff
@@ -58,51 +64,43 @@ fun buildPreparableFiltersSql(
         metadataOptions: Set<MetadataOption>,
         linking: Boolean,
         idsPresent: Boolean,
-        partitionsPresent: Boolean
+        partitionsPresent: Boolean,
+        detailed: Boolean = false
 ): Pair<String, Set<SqlBinder>> {
     val filtersClauses = buildPreparableFiltersClause(startIndex, propertyTypes, propertyTypeFilters)
     val filtersClause = if (filtersClauses.first.isNotEmpty()) " AND ${filtersClauses.first} " else ""
-
     val metadataOptionColumns = metadataOptions.associateWith(::mapMetaDataToColumnSql)
-    val nonAggregatedMetadataSql = metadataOptionColumns
-            .filter { !isMetaDataAggregated(it.key) }.values.joinToString("")
-    val innerGroupBy = groupBy(ESID_EKID_PART_PTID + nonAggregatedMetadataSql)
-    // TODO: remove IS NOT NULL post-migration
-    val linkingClause = if (linking) {
-        " AND ${ORIGIN_ID.name} IS NOT NULL AND ${ORIGIN_ID.name} != '${IdConstants.EMPTY_ORIGIN_ID.id}' "
-    } else {
-        ""
-    }
+    val metadataOptionColumnsSql = metadataOptionColumns.values.joinToString("")
+    val innerGroupBy = groupBy(ESID_EKID_PART_PTID)
+    val outerGroupBy = if (metadataOptions.contains(MetadataOption.ORIGIN_IDS)) {
```

Comment:
`PROPERTIES` should be in group by or aggregated in outer query

Reply:
Missed a jsonb_object_agg